RPG Wizardry/Assets/Scripts/UI/GameUI/GameUIManager.cs

```diff
@@ -36,15 +71,27 @@ private void Update()
         /// </summary>
         private void CheckPlayerInput()
         {
-            if (Input.GetKeyDown(KeyCode.Escape))
+            if (PauseAllowed && Input.GetKeyDown(KeyCode.Escape)) //Added the Pause Allowed check
             {
                 GameManager.Instance.TogglePause();
                 if (GameManager.Instance.Paused) // Game was running, open Menu
                     SceneLoader.Instance.LoadMenuScene();
                 else // Game was Paused. Close Menu
                     SceneLoader.Instance.LoadGameScene();
             }
+        }
+        #endregion
+
+        #region Public
+
+        /// <summary>
+        /// Toggles the ESC button value (true and false) using the Pauseallowed Event
+        /// </summary>
+        public void ToggelPause(bool value)

```

Comment:
As .Update is only used to check for the Pause-Button, this functionality can also be implemented using Unity's MonoBehaviour.enabled-tools.